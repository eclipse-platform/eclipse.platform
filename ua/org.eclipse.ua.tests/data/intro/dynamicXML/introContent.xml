<?xml version="1.0" encoding="utf-8" ?>
<!--
     Copyright (c) 2005, 2006 IBM Corporation and others.

     This program and the accompanying materials
     are made available under the terms of the Eclipse Public License 2.0
     which accompanies this distribution, and is available at
     https://www.eclipse.org/legal/epl-2.0/

     SPDX-License-Identifier: EPL-2.0
    
     Contributors:
         IBM Corporation - initial API and implementation
 -->

<!--
   An intro for testing dynamic content in intro XML. This is a very simple intro with no
   images or graphics that has filtering, includes and anchor examples.
-->
<introContent>

	<!--
	   The main page, containing links to the various examples.
	-->
    <page id="root">
        <title>Dynamic XML Intro</title>
        <group>
            <link label="Static Content Example" url="http://org.eclipse.ui.intro/showPage?id=static"/>
        </group>
        <group>
            <link label="Filter Example" url="http://org.eclipse.ui.intro/showPage?id=filter"/>
        </group>
        <group>
            <link label="Include Example" url="http://org.eclipse.ui.intro/showPage?id=include"/>
        </group>
        <group>
            <link label="Anchor Example" url="http://org.eclipse.ui.intro/showPage?id=anchor"/>
        </group>
    </page>

	<!--
	   A simple static page that shows one link, no dynamic content. Used as a sanity test.
	-->
    <page id="static">
        <title>Static Content Example</title>
        <group>
            <link label="This page contains static content only." url="http://org.eclipse.ui.intro/showPage?id=static"/>
        </group>
    </page>

	<!--
	   The filter examples page. Has links that are os/ws/arch/etc specific. When viewed, this page
	   should state facts about your environment. For exmaple, your OS is win32, your WS is win32, etc.
	-->
    <page id="filter">
        <title>Filter Example</title>
        <group>
            <link label="Sanity test" url="http://org.eclipse.ui.intro/showPage?id=filter"/>

            <!-- should never appear -->
            <link filter="os=invalid" label="Your OS is invalid." url="http://org.eclipse.ui.intro/showPage?id=filter"/>
            <link filter="ws=invalid" label="Your WS is invalid." url="http://org.eclipse.ui.intro/showPage?id=filter"/>
            <link filter="arch=invalid" label="Your ARCH is invalid." url="http://org.eclipse.ui.intro/showPage?id=filter"/>
            <link label="Your OS is win32 AND your WS is invalid." url="http://org.eclipse.ui.intro/showPage?id=filter">
                <filter name="os" value="win32"/>
                <filter name="ws" value="invalid"/>
            </link>
        </group>
        <group>
            <link filter="plugin=org.eclipse.help" label="The plugin org.eclipse.help is installed." url="http://org.eclipse.ui.intro/showPage?id=filter"/>

            <!-- should never appear -->
            <link filter="plugin=invalid" label="The plugin invalid is installed." url="http://org.eclipse.ui.intro/showPage?id=filter"/>
        </group>
        <group>
            <link filter="product=org.eclipse.sdk.ide" label="The product org.eclipse.sdk.ide is running." url="http://org.eclipse.ui.intro/showPage?id=filter"/>

            <!-- should never appear -->
            <link filter="plugin=invalid" label="The product invalid is running." url="http://org.eclipse.ui.intro/showPage?id=filter"/>
        </group>
        <group>
            <link filter="category=org.eclipse.categories.developmentCategory" label="The category org.eclipse.categories.developmentCategory is enabled." url="http://org.eclipse.ui.intro/showPage?id=filter"/>

            <!-- should never appear -->
            <link filter="category=invalid" label="The category invalid is enabled." url="http://org.eclipse.ui.intro/showPage?id=filter"/>
        </group>
        <group>
            <link filter="activity=org.eclipse.javaDevelopment" label="The activity org.eclipse.javaDevelopment is enabled." url="http://org.eclipse.ui.intro/showPage?id=filter"/>

            <!-- should never appear -->
            <link filter="activity=invalid" label="The activity invalid is enabled." url="http://org.eclipse.ui.intro/showPage?id=filter"/>
        </group>
    </page>

	<!--
	   The include examples page. The include should bring in the group from the page include2 below.
	   There should be two links on this page when viewed.
	-->
    <page id="include">
        <title>Include Example</title>
        <group>
            <link label="There should be one link below included from another page." url="http://org.eclipse.ui.intro/showPage?id=include"/>
        </group>
        <include path="include2/my.id.include2"/>
    </page>

	<!--
	   You never actually see this page, it is only used to define a group that is included in page
	   "include" above.
	-->
    <page id="include2">
        <title>Include Example 2</title>
        <group id="my.id.include2">
            <link label="This link is from page id=include2 and is to be included into other pages." url="http://org.eclipse.ui.intro/showPage?id=include"/>
        </group>
    </page>

	<!--
	   The anchor examples page. This page contains an anchor. The content is contributed from an intro
	   extension. See ext.xml in the same folder.
	-->
    <page id="anchor">
        <title>Anchor Example</title>
        <group>
            <link label="There is an anchor below this link that should be populated from another page." url="http://org.eclipse.ui.intro/showPage?id=anchor"/>
        </group>
        <anchor id="my.id.anchor"/>
    </page>

</introContent>
