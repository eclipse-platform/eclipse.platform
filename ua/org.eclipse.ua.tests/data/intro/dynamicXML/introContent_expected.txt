Intro Model Content:
======================

Model has valid config = true
Presentation type = 2
Home page id = root
Number of pages (not including Root Page) = 5
Number of shared groups = 0
Number of unresolved extensions = 0

HOME PAGE: 
--------------
	is dynamic= true
	id = root
	title = Dynamic XML Intro
	style = null
	alt-style = null
	url = null
	style-id = null
	page styles are = 
	page alt-styles are = 
	page children = 5

		TITLE: id = null
				title = Dynamic XML Intro
				style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = Static Content Example
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=static
								style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = Filter Example
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=filter
								style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = Include Example
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=include
								style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = Anchor Example
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=anchor
								style-id = null

PAGE id = static
----------
	title = Static Content Example
	style = null
	alt-style = null
	style-id = null
	page styles are = 
	page alt-styles are = 
	page children = 2

		TITLE: id = null
				title = Static Content Example
				style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = This page contains static content only.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=static
								style-id = null

PAGE id = filter
----------
	title = Filter Example
	style = null
	alt-style = null
	style-id = null
	page styles are = 
	page alt-styles are = 
	page children = 6

		TITLE: id = null
				title = Filter Example
				style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = Sanity test
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=filter
								style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = The plugin org.eclipse.help is installed.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=filter
								style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = The product org.eclipse.sdk.ide is running.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=filter
								style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = The category org.eclipse.categories.developmentCategory is enabled.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=filter
								style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = The activity org.eclipse.javaDevelopment is enabled.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=filter
								style-id = null

PAGE id = include
----------
	title = Include Example
	style = null
	alt-style = null
	style-id = null
	page styles are = 
	page alt-styles are = 
	page children = 3

		TITLE: id = null
				title = Include Example
				style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = There should be one link below included from another page.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=include
								style-id = null
		GROUP: id = my.id.include2
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = This link is from page id=include2 and is to be included into other pages.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=include
								style-id = null

PAGE id = include2
----------
	title = Include Example 2
	style = null
	alt-style = null
	style-id = null
	page styles are = 
	page alt-styles are = 
	page children = 2

		TITLE: id = null
				title = Include Example 2
				style-id = null
		GROUP: id = my.id.include2
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = This link is from page id=include2 and is to be included into other pages.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=include
								style-id = null

PAGE id = anchor
----------
	title = Anchor Example
	style = null
	alt-style = null
	style-id = null
	page styles are = 
	page alt-styles are = 
	page children = 4

		TITLE: id = null
				title = Anchor Example
				style-id = null
		GROUP: id = null
				label = null
				children = 1
				style-id = null
						LINK: id = null
								label = There is an anchor below this link that should be populated from another page.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=anchor
								style-id = null
		GROUP: id = null
				label = null
				children = 2
				style-id = null
						LINK: id = null
								label = This link is from extension content ext.xml and is to be contributed into a page with an anchor.
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=anchor
								style-id = null
						LINK: id = null
								label = The plug-in org.eclipse.help is installed
								text = null
								url = http://org.eclipse.ui.intro/showPage?id=anchor
								style-id = null
		ANCHOR: id = my.id.anchor

Model Flag Tests: 
----------------
		First page children are: 
			Groups: 1
			Links: 0
			Texts: 0
			HTMLs: 0
			Images: 0
			Includes: 0
			Page Titles: 1
			Page Heads: 0
			Model Elements: 2
			Containers: 1
			All Pages: 0
			Anchors: 0
			Content providers: 0
			Elements with Text child(AbstractTextElemets): 0
			Groups and Links: 1